data/mp/eda_mp_trj.py

```diff
@@ -77,22 +86,29 @@ def info_dict_to_id(info: dict[str, int | str]) -> str:
 df_mp_trj = pd.DataFrame(
     {
         info_dict_to_id(atoms.info): atoms.info
-        | {key: atoms.arrays.get(key) for key in ("forces", "magmoms")}
-        | {"formula": str(atoms.symbols), Key.atom_nums: atoms.symbols}
+        | {
+            Key.forces: atoms.get_forces(),
+            Key.stress: atoms.get_stress(),
+            Key.magmoms: atoms.get_magnetic_moments()
+            if "magmoms" in atoms.calc.results
+            else None,
+            Key.formula: str(atoms.symbols),
+            Key.atom_nums: atoms.symbols,
+        }
         for atoms_list in tqdm(mp_trj_atoms.values(), total=len(mp_trj_atoms))
         for atoms in atoms_list
     }
 ).T.convert_dtypes()  # convert object columns to float/int where possible
 df_mp_trj.index.name = "frame_id"
-assert len(df_mp_trj) == 1_580_312  # number of total frames
+assert len(df_mp_trj) == 1_580_395  # number of total frames
 if Key.formula not in df_mp_trj:
     raise KeyError(f"{Key.formula!s} not in {df_mp_trj.columns=}")
 
 # this is the unrelaxed (but MP2020 corrected) formation energy per atom of the actual
 # relaxation step
 df_mp_trj = df_mp_trj.rename(columns={"ef_per_atom": MbdKey.e_form_dft})
```

Comment:
@CompRhys reran the script with the new fixed `2024-09-03-mp-trj.extxyz.zip`. all cells pass after 9058aa7 except 

```py
fig = pmv.histogram(df_mp_trj[MbdKey.e_form_dft], bins=300)
```

since the DFT formation energy is no longer recorded in the `Atoms.info` dict

Reply:
When chunked the script runs fairly quickly, two options exist: i) rerun that script to keep the extra labels I didn't save, I think I didn't do this as some of the formation energies looked wrong to me as they were the same as the energy per atom. ii) just grab the reference energies for the compositions and compute it quickly with https://github.com/janosh/matbench-discovery/blob/4d3d33f63a963f0d54f5538af04dd74866456c8f/matbench_discovery/energy.py#L83C5-L83C24

Reply:
yeah, leaning towards ii) as it seems slightly less reliant on flawless data provenance

Reply:
option ii) implemented in [a1f92db](https://github.com/janosh/matbench-discovery/pull/130/commits/a1f92db1c93e3b57ad6037a1782e419be6401371) and confirmed to yield identical results as before